src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -250,13 +250,12 @@ impl<'a, 'tcx> PhiInserter<'a, 'tcx> {
     fn pack(mut self) -> ykpack::Pack {
         let mut df = DominatorFrontiers::new(self.mir);
         let num_tir_vars = self.def_sites.len();
-        let num_tir_blks = self.mir.basic_blocks().len(); // There's one TIR block per MIR block.
 
         // We first need a mapping from block to the variables it defines. Appel calls this
         // `A_{orig}`. We can derive this from our definition sites.
-        let a_orig = {
+        let (a_orig, num_tir_blks) = {
             let ykpack::Pack::Mir(ykpack::Mir{ref blocks, ..}) = self.pack;
-            debug_assert!(num_tir_blks == blocks.len());
```

Comment:
Here's where that assertion is removed.

The reason I did it this way was because, at some point we may move away from one MIR block per TIR block. In that case, this code remains correct.

[I'm hoping we don't have to make that move, as it would mean we'd have to implement our own dominators and dominator tree structures, instead of relying on the ones from the compiler, but that's another story]

Reply:
I'm not sure I understand, but it's not that important either way.